web/src/compose_validate.ts

```diff
@@ -806,18 +806,49 @@ export function validate_message_length($container: JQuery): boolean {
     if (text.length > realm.max_message_length) {
         $textarea.addClass("flash");
         setTimeout(() => $textarea.removeClass("flash"), 1500);
+
+        const $banner_container = $("#compose_banners");
+        compose_banner.show_error_message(
+            $t({defaultMessage: "Message length shouldn't be greater than 10000 characters."}),
+            compose_banner.CLASSNAMES.missing_private_message_recipient,
+            $banner_container,
+            $("#private_message_recipient"),
+        );
+
         return false;
     }
     return true;
 }
 
 export function validate(scheduling_message: boolean): boolean {
     const message_content = compose_state.message_content();
+    // P1 and P2
```

Comment:
Instead of writing "P1", "P2", etc, I think it would be clearer to just have a comment at the top here that says:

```
// The validation checks in this function are in a specific priority order. Don't
// change their order unless you want to change which priority they're shown in.
```